src/views/Log.vue

```diff
@@ -264,34 +295,59 @@ export default {
     }
   },
 
+  emits: [
+    updateInitialOptionsEvent,
+  ],
+
```

Comment:
in the other views I think the initialOptions logic is inside src/components/cylc/`view-name`/`view-name`.vue
Should this be the case for log view as well for consistency?

Reply:
It is the views (in `src/views/`) that we mount in tabs in the workspace, so these are where we want the `initialOptions` prop to live.

You have correctly pointed out that in some cases, the component (in `src/components`) contains the state we want to save per view. We might need to refactor those to extract the state into the view, which would be passed into the component as a prop (probably using `v-model`).

Reply:
On the split between `src/views` and `src/components`. IMO:

Views are cylc-ui plugins, the view code should contain all the boiler plate for the plugin, e.g. the views name & icon, the logic for subscription handling, mappings onto the data store, user facing options and their defaults.

Components are generic reusable items that could, conceivably, find application in multiple views. They should contain the logic for their own rendering but be as ignorant as possible of the cylc-ui application itself (but will have to make assumptions about the format of the data they are provided).